code/modules/mining/lavaland/loot/tendril_loot.dm

```diff
@@ -174,9 +176,37 @@
 	if(istype(next, /turf/simulated/floor/plating/lava/smooth) || istype(current, /turf/simulated/floor/plating/lava/smooth)) //We can move from land to lava, or lava to land, but not from land to land
 		..()
 	else
-		to_chat(user, "<span class='warning'>Boats don't go on land!</span>")
+		if(last_message_time + 1 SECONDS < world.time)
+			to_chat(user, "<span class='warning'>Boats don't go on land!</span>")
+			last_message_time = world.time
 		return FALSE
 
+/obj/vehicle/lavaboat/Destroy()
+	for(var/mob/living/M in buckled_mobs)
+		M.weather_immunities -= "lava"
+		UnregisterSignal(M, COMSIG_MOVABLE_MOVED)
+	. = ..()
+
+// failsafe, if for some reason we aren't on the same turf (teleport or something)
+/obj/vehicle/lavaboat/proc/on_user_move(mob/living/moved, atom/oldloc, direction)
```

Comment:
I noticed that when a mob teleports, it should already call `unbuckle_mob` on whatever it's buckled to. I'm wondering if the signal stuff is necessary here.

Reply:
Good call, I now see it's called in `forceMove`. I'll remove the signal stuff. ~~Now just hope nobody's setting .locs~~